src/debug/di/CMakeLists.txt

```diff
@@ -32,51 +36,50 @@ set(CORDBDI_SOURCES
+        convert_to_absolute_path(CORDBDI_SOURCES_ASM_FILE ${CORDBDI_SOURCES_ASM_FILE})
 
-          # We do not pass any defines since we have already done pre-processing above
-          set (ASM_CMDLINE "-o ${CMAKE_CURRENT_BINARY_DIR}/${name}.obj ${ASM_PREPROCESSED_FILE}")
+        # Inserts a custom command in CMake build to preprocess each asm source file
+        get_filename_component(name ${CORDBDI_SOURCES_ASM_FILE} NAME_WE)
+        file(TO_CMAKE_PATH "${CMAKE_CURRENT_BINARY_DIR}/${name}.asm" ASM_PREPROCESSED_FILE)
+        preprocess_def_file(${CORDBDI_SOURCES_ASM_FILE} ${ASM_PREPROCESSED_FILE})
 
-          # Generate the batch file that will invoke the assembler
-          file(TO_CMAKE_PATH "${CMAKE_CURRENT_BINARY_DIR}/runasm_${name}.cmd" ASM_SCRIPT_FILE)
+        # We do not pass any defines since we have already done pre-processing above
+        set (ASM_CMDLINE "-o ${CMAKE_CURRENT_BINARY_DIR}/${name}.obj ${ASM_PREPROCESSED_FILE}")
 
-          file(GENERATE OUTPUT "${ASM_SCRIPT_FILE}"
-               CONTENT "\"${CMAKE_ASM_MASM_COMPILER}\" -g ${ASM_CMDLINE}")
+        # Generate the batch file that will invoke the assembler
+        file(TO_CMAKE_PATH "${CMAKE_CURRENT_BINARY_DIR}/runasm_${name}.cmd" ASM_SCRIPT_FILE)
 
-          message("Generated  - ${ASM_SCRIPT_FILE}")
+        file(GENERATE OUTPUT "${ASM_SCRIPT_FILE}"
+            CONTENT "\"${CMAKE_ASM_MASM_COMPILER}\" -g ${ASM_CMDLINE}")
 
-          # Need to compile asm file using custom command as include directories are not provided to asm compiler
-          add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${name}.obj
-                             COMMAND ${ASM_SCRIPT_FILE}
-                             DEPENDS ${ASM_PREPROCESSED_FILE}
-                             COMMENT "Assembling ${ASM_PREPROCESSED_FILE} - ${ASM_SCRIPT_FILE}")
+        message("Generated  - ${ASM_SCRIPT_FILE}")
 
-          # mark obj as source that does not require compile
-          set_source_files_properties(${CMAKE_CURRENT_BINARY_DIR}/${name}.obj PROPERTIES EXTERNAL_OBJECT TRUE)
+        # Need to compile asm file using custom command as include directories are not provided to asm compiler
+        add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${name}.obj
+                            COMMAND ${ASM_SCRIPT_FILE}
+                            DEPENDS ${ASM_PREPROCESSED_FILE}
+                            COMMENT "Assembling ${ASM_PREPROCESSED_FILE} - ${ASM_SCRIPT_FILE}")
 
-          # Add the generated OBJ in the dependency list so that it gets consumed during linkage
-          set(CORDBDI_SOURCES ${CORDBDI_SOURCES} ${CMAKE_CURRENT_BINARY_DIR}/${name}.obj)
+        # mark obj as source that does not require compile
+        set_source_files_properties(${CMAKE_CURRENT_BINARY_DIR}/${name}.obj PROPERTIES EXTERNAL_OBJECT TRUE)
+        # undefine CORDBDI_SOURCES_ASM_FILE to prevent incusion in project
+        unset(CORDBDI_SOURCES_ASM_FILE)
```

Comment:
Unfortunately, it seems that there is no CMake source file property which could be used to prevent compilation errors with asm file requiring preprocessing. CMake classifies passed source files depending on their extension and asm file could either be passed as object (best option but such file is than passed as object file to linker) or we could set any other available property but none of them will prevent compilation.

Decided to unset variable to prevent passing it into CMake.

Reply:
I am fine with this solution too, but I don't understand how come changing this:
```cmake
set(CORDBDI_SOURCES ${CORDBDI_SOURCES} ${CMAKE_CURRENT_BINARY_DIR}/${name}.obj)
```
to this:
```cmake
set(CORDBDI_SOURCES_ASM_FILE ${CMAKE_CURRENT_BINARY_DIR}/${name}.obj)
.
.
.
list(APPEND CORDBDI_SOURCES ${CORDBDI_SOURCES_ASM_FILE}) 
```
would not work. Basically, it is just postponing adding the .obj to the CORDBDI_SOURCES.

Reply:
Yup, I have overlooked this, it happened due to not removing carefully enough WIP edits. Will correct.

Reply:
@janvorli 

```
set(CORDBDI_SOURCES_ASM_FILE ${CMAKE_CURRENT_BINARY_DIR}/${name}.obj)
```

This code does not exist in current version. Double checked code and I think that major problem is caused by messy treatment of white space changes by git/github what makes reading code changes very hard even for me who made this changes.

As you indicated the change essentially postpones adding `asm` file itself until after `precompiled headers` are added and does not change treatment of `obj` file generated by `masm`. In the case of building for `arm` and `arm64` Windows targets it means that `asm` files are preprocessed, compiled to `obj` file and passed with sources with added property `EXTERNAL_OBJECT` indicating that `obj` should be used during linking as lib.

In the case of cross component build on x86 Win machine the asm files for `arm` or `arm64` are not processed at all.

Reply:
> This code does not exist in current version.

I am sorry for not being clear. I meant that you would actually make the change from 
```cmake
set(CORDBDI_SOURCES ${CORDBDI_SOURCES} ${CMAKE_CURRENT_BINARY_DIR}/${name}.obj)
``` 
to 
```cmake
set(CORDBDI_SOURCES_ASM_FILE ${CMAKE_CURRENT_BINARY_DIR}/${name}.obj)
```

And then the unconditional
```cmake
list(APPEND CORDBDI_SOURCES ${CORDBDI_SOURCES_ASM_FILE}) 
```
Would handle adding the .obj file too.

Reply:
Agree, I would prefer to add this change in one of the other PRs as separate commit as CI tests which are usually failing at start due to full disk are actually running and perhaps will pass 😄 If not, I would update this PR.